fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/event/lifecycle/v1/ServerLifecycleCallback.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.lifecycle.v1;
+
+import net.minecraft.server.MinecraftServer;
+
+/**
+ * A callback which is invoked when the server advances in it's lifecycle.
+ */
+public interface ServerLifecycleCallback {
+	void onChangeLifecycle(MinecraftServer server);
```

Comment:
Maybe `lifecycle`?

Reply:
Well this means the server's lifecycle stage has changed. onLifecycle is a bit vague it would somewhat correspond to a `lifecycle` method

Reply:
`lifecycle` is a verb. It has `cycle` meaning it is cycling to a next stage.

Reply:
Sorry what ? https://www.merriam-webster.com/dictionary/life%20cycle

Reply:
Gradle has a `lifecycle` that is like a `info` for logging. This current name is acceptable.